distutils/extension.py

```diff
@@ -106,16 +108,22 @@ def __init__(
     ):
         if not isinstance(name, str):
             raise AssertionError("'name' must be a string")  # noqa: TRY004
-        if not (
-            isinstance(sources, list)
-            and all(isinstance(v, (str, os.PathLike)) for v in sources)
-        ):
+
+        # we handle the string case first; though strings are iterable, we disallow them
```

Comment:
nitpick: Try to use imperative voice in comments.

```suggestion
        # handle the string case first; since strings are iterable, disallow them
```

Reply:
Thanks!